sklearn/mixture/tests/test_gaussian_mixture.py

```diff
@@ -0,0 +1,681 @@
+    for cv_type in COVARIANCE_TYPE:
+        g = GaussianMixture(
+            n_components=n_components, covariance_type=cv_type,
+            random_state=rng)
+        g.n_features = n_features
+        assert_equal(g._n_parameters(), n_params[cv_type])
+
+
+def test_gaussian_mixture_aic_bic():
+    # Test the aic and bic criteria
+    rng = np.random.RandomState(0)
+    n_samples, n_features, n_components = 50, 3, 2
+    X = rng.randn(n_samples, n_features)
+    # standard gaussian entropy
+    SGH = 0.5 * (np.log(np.linalg.det(np.cov(X.T, bias=1))) +
+                 n_features * (1 + np.log(2 * np.pi)))
+    for cv_type in COVARIANCE_TYPE:
+        g = GaussianMixture(
+            n_components=n_components, covariance_type=cv_type,
+            random_state=rng)
+        g.fit(X)
+        aic = 2 * n_samples * SGH + 2 * g._n_parameters()
+        bic = (2 * n_samples * SGH +
+               np.log(n_samples) * g._n_parameters())
+        bound = n_features * 3. / np.sqrt(n_samples)
+        assert_true(np.abs(g.aic(X) - aic) / n_samples < bound)
+        assert_true(np.abs(g.bic(X) - bic) / n_samples < bound)
+
+
+def test_gaussian_mixture_verbose():
+    rng = np.random.RandomState(0)
+    rand_data = RandomData(rng)
+    n_components = rand_data.n_components
+    for cov_type in COVARIANCE_TYPE:
+        X = rand_data.X[cov_type]
+        g = GaussianMixture(n_components=n_components, n_init=1,
+                            max_iter=100, reg_covar=0, random_state=rng,
+                            covariance_type=cov_type, verbose=1)
+        h = GaussianMixture(n_components=n_components, n_init=1,
+                            max_iter=100, reg_covar=0, random_state=rng,
+                            covariance_type=cov_type, verbose=2)
+        k = GaussianMixture(n_components=n_components, n_init=1,
+                            max_iter=100, reg_covar=0, random_state=rng,
+                            covariance_type=cov_type, verbose=20)
+        old_stdout = sys.stdout
+        sys.stdout = StringIO()
+        try:
+            g.fit(X)
+            h.fit(X)
+            k.fit(X)
+        finally:
+            sys.stdout = old_stdout
+
+
+def test_warm_start():
+    rng = np.random.RandomState(0)
+    rand_data = RandomData(rng)
+
+    X = rand_data.X['full']
+
+    rng = np.random.RandomState(0)
+    g = GaussianMixture(n_components=rand_data.n_components, n_init=1,
+                        max_iter=100, reg_covar=0, random_state=rng,
+                        warm_start=False)
+    rng = np.random.RandomState(0)
+    h = GaussianMixture(n_components=rand_data.n_components, n_init=1,
+                        max_iter=50, reg_covar=0, random_state=rng,
+                        warm_start=True)
+
+    g.fit(X)
+    h.fit(X)
+    h.fit(X)
+
+    assert_almost_equal(g.weights_, h.weights_)
+    assert_almost_equal(g.means_, h.means_)
+    assert_almost_equal(g.covariances_, h.covariances_)
```

Comment:
regarding tests 

lines 398, 424 are not covered.

I don't see any test of score and score_samples methods